tracing/otel_test.go

```diff
@@ -0,0 +1,146 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace/noop"
+)
+
+func TestParseAttributes(t *testing.T) {
+	os.Setenv("EXISTENT_ENV_KEY", "env_value")
+	defer os.Unsetenv("EXISTENT_ENV_KEY")
+	t.Run("ValidAttributes", func(t *testing.T) {
+		tests := []struct {
+			input          string
+			expectedOutput []attribute.KeyValue
+			expectedError  error
+		}{
+			{
+				input: "key1=value1,key2=value2",
+				expectedOutput: []attribute.KeyValue{
+					attribute.String("key1", "value1"),
+					attribute.String("key2", "value2"),
+				},
+				expectedError: nil,
+			},
+			{
+				input: "key1=${EXISTENT_ENV_KEY},key2=${NON_EXISTENT_ENV_KEY:default_value}",
+				expectedOutput: []attribute.KeyValue{
+					attribute.String("key1", os.Getenv("EXISTENT_ENV_KEY")),
```

Comment:
nit: let's just hardcode expected value
```suggestion
					attribute.String("key1", "env_value"),
```

Reply:
Done in https://github.com/grafana/dskit/pull/681/commits/b8e1ea2bcc3e176d9517bcb861fd1bba2b0a5e99